File: cracking-the-coding-interview-4/01.1/test_1.pl
$string="asdf";

@string=split /\s*/,$string;

foreach $a (@string) {
	$a{$a}++;
}

foreach $a (sort keys(%a)) {
	# print "$a: $a{$a}\n";
	if ($a{$a}>1) {
		print "Duplicates present\n";
		 exit;
	}
}
print "No Duplicates\n";
exit;

File: cracking-the-coding-interview-4/01.1/test_1.py
import sys
from collections import defaultdict

string="asdf";

a=defaultdict(int)

for ii in string:
	a[ii]+=1

for vv in a.itervalues():
	if vv > 1:
		print "Duplicates present"
		sys.exit()

print "No Duplicates"
sys.exit()

